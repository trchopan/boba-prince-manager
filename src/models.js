export const profileParser = data => ({
  fullname: data.fullname || "",
  position: data.position || "user",
  avatar: data.avatar || "/images/bloody-smile.jpg"
});
