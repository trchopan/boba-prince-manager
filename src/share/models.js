export const themes = {
  light: "Light",
  dark: "Dark"
};

export const languages = {
  vi: { code: "vi", name: "Tiáº¿ng Viá»‡t", char: "ðŸ‡»ðŸ‡³" },
  en: { code: "en", name: "English", char: "ðŸ‡¬ðŸ‡§" }
};

export const profileParser = data => ({
  fullname: data.fullname || "",
  position: data.position || "user",
  avatar: data.avatar || "/images/bloody-smile.jpg"
});
