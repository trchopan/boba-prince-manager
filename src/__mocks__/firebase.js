export const fakeAuthUser = { uid: "123456", email: "tester@arkham.city" };
export const fakeError = { code: "fake-error" };
export const docData = { name: "tester" };

const docResult = {
  exists: true,
  data: () => docData
};

const collection = jest
  .fn()
  .mockImplementationOnce(() => ({
    doc: () => ({
      get: () => Promise.resolve(docResult)
    })
  }))
  .mockImplementationOnce(() => ({
    doc: () => ({
      get: () => Promise.reject(fakeError)
    })
  }));

export const fireStore = { collection };

const signInWithEmailAndPassword = jest
  .fn()
  .mockImplementationOnce(() => Promise.resolve())
  .mockImplementationOnce(() => Promise.reject(fakeError));
const signOut = jest
  .fn()
  .mockImplementationOnce(() => Promise.resolve())
  .mockImplementationOnce(() => Promise.reject(fakeError));
export const fireAuth = { signInWithEmailAndPassword, signOut };
