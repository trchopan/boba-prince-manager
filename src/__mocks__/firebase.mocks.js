export const fakeSignUpCredentials = {
  email: "test@tester.com",
  password: "123456",
  response: "abcxyz123"
};
export const fakeAuthUser = { uid: "123456", email: "tester@arkham.city" };
export const fakeError = { code: "fake-error" };
export const docData = {
  fullname: "Tester",
  avatar: "avatar",
  position: "manager"
};

export const createSnapshot = (data) => ({
  exists: true,
  data: () => data
});

export const createFirestoreOnSnapshot = (docSnapshot) => ({
  collection: () => ({
    doc: () => ({
      onSnapshot: (cb, _) => cb(docSnapshot)
    })
  }) 
})

export const createFirestoreOnSnapshotError = (error) => ({
  collection: () => ({
    doc: () => ({
      onSnapshot: (_, err) => err(error)
    })
  }) 
})
