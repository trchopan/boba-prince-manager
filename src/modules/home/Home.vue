<template>
  <div>
    <my-header>{{ $t.home }}</my-header>
    <div v-if="!authUser" class="home-container">
      <p>{{ $t.notYetHasAccount }}</p>
      <v-btn to="/signup" outline color="secondary">{{ $t.pressHereToSignUp }}</v-btn>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex";

export const languagesMap = {
  home: { vi: "Troang chủ", en: "Home page" },
  notYetHasAccount: {
    vi: "Bạn chưa có tài khoản?",
    en: "Not yet have account?"
  },
  pressHereToSignUp: {
    vi: "Hãy nhấn vào đây để đăng ký",
    en: "Click here to sign up"
  }
};

export default {
  name: "Home",
  computed: {
    ...mapGetters({
      language: "layout/language",
      authUser: "auth/authUser"
    }),
    $t() {
      return this.$translate(languagesMap, this.language.value);
    }
  }
};
</script>

<style lang="scss" scoped>
.home-container {
  text-align: center;
}
</style>
