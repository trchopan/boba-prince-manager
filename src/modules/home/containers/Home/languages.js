export const LANGUAGES_MAP = {
  greeting: {
    vi: "Lời chào",
    en: "Greeting"
  },
  hello: {
    vi: "Xin chào, tên tôi là",
    en: "Hi, My name is"
  },
  name: {
    vi: "Trần Quang (tên khác Chop Tr)",
    en: "Quang Tran (aka Chop Tr)"
  },
  greetingText: {
    vi:
      "Lập trình đối với tôi là một sở thích từ nhỏ nhưng dần dần đã trở thành một niềm đam mê mạnh mẽ. \
      Tính tôi thích logic và quy cách, cấu trúc, tôi có thể thỏa thích trải nghiệm và tìm tòi những thứ mới mỗi khi đắm mình vào những dòng mã nguồn. \
      Và trang web này phục vụ như là demo những kỹ năng của tôi học được và là sân chơi cho những thứ tôi đang tìm tòi.",
    en:
      "Software Engineer at first is just a hobby of mine but eventually it became my fuel and fire. \
      I can satisfy the need for good structure design and to thrive for new thing everytime I begin to flow in the code. \
      This website serves as a demonstration of my knowledge and skill. \
      It's also a playground for things I am learning."
  },
  skillLevel: {
    vi: "Trình độ",
    en: "Skill level"
  },
  fieldsOfInterest: {
    vi: "Lĩnh vực đang quan tâm",
    en: "Fields of interest"
  }
};
