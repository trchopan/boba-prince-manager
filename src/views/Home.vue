<template>
  <div class="home">
    <p>Xin chào quản lý, phẻ hok?</p>
  </div>
</template>

<script>
export default {
  name: "home",
  methods: {
    logout() {
      this.$store.dispatch("auth/logout");
    }
  }
};
</script>
