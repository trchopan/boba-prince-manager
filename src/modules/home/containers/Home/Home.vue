<template>
  <div>
    <section id="home-header">
      <h1 class="display-1 primary--text">{{ $t.greeting }}</h1>
      <p>
        {{ $t.hello }}
        <span
          class="font-weight-bold secondary--text"
        >{{ $t.name }}</span>
      </p>
      <p>{{ $t.greetingText }}</p>
    </section>
    <section id="home-playgrounds">
      <h2 class="headline primary--text mb-3">{{ $t.skillLevel }}</h2>
      <div
        v-for="skill in skills"
        :key="'skill-' + skill.title"
        class="mb-3"
      >
        <div class="title secondary--text mb-3">
          <router-link :to="skill.route">{{ skill.title }}</router-link>
        </div>
        <v-rating
          readonly
          :value="skill.level"
          color="primary"
        ></v-rating>
        <div v-if="skill.interests.length > 0">
          <p class="subheading">{{ $t.fieldsOfInterest }}</p>
          <ul>
            <li
              v-for="interest in skill.interests"
              :key="'interest-' + interest.name"
            >
              {{ interest.name }}
              <v-icon
                small
                @click="open(interest.link, true)"
              >open_in_new</v-icon>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import { LANGUAGES_MAP } from "./languages.js";

export default {
  name: "Home",
  computed: {
    ...mapGetters({
      language: "layout/language",
      authUser: "auth/authUser"
    }),
    $t() {
      return this.$translate(LANGUAGES_MAP, this.language.value);
    },
    skills() {
      return [
        {
          title: this.$t.vue,
          route: "/playground/vue",
          level: 5,
          interests: [
            {
              name: this.$t.vueNuxt,
              link: "https://nuxtjs.org/"
            },
            {
              name: this.$t.vueTypescript,
              link: "https://vuejs.org/v2/guide/typescript.html"
            }
          ]
        },
        {
          title: this.$t.firebase,
          route: "/playground/firebase",
          level: 5,
          interests: [
            {
              name: "Firebase ML Kit",
              link: "https://firebase.google.com/products/ml-kit/"
            }
          ]
        },
        {
          title: this.$t.angular6,
          route: "/playground/angular",
          level: 5,
          interests: [
            { name: this.$t.angular6Nwrl, link: "https://nrwl.io/nx" }
          ]
        },
        {
          title: this.$t.react,
          route: "/playground/react",
          level: 4,
          interests: [
            {
              name: this.$t.reactHooks,
              link: "https://reactjs.org/docs/hooks-intro.html"
            }
          ]
        }
      ];
    }
  },
  methods: {
    open(link, external) {
      window.open(link, external ? "_blank" : "");
    }
  }
};
</script>

<style lang="scss" scoped>
section > * {
  text-align: justify;
}
a {
  color: inherit;
}
</style>
