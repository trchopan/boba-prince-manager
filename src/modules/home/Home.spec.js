import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Translation from "@/plugins/translation.js";
import Home from "./Home.vue";
import { languages } from "@/modules/core/layout.models.js";
import { mockCustomElements } from "@/__mocks__/custom-elements.js";
import { fakeAuthUser } from "@/__mocks__/firebase-results";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Translation);

describe("Home.vue", () => {
  let getters;
  let store;
  let wrapper;

  it("matchs snapshot", () => {
    getters = {
      "auth/authUser": () => fakeAuthUser,
      "layout/language": () => languages.en
    };
    store = new Vuex.Store({
      getters
    });
    wrapper = shallowMount(Home, {
      stubs: mockCustomElements,
      store,
      localVue
    });
    expect(wrapper).toMatchSnapshot();

    getters = {
      ...getters,
      "auth/authUser": () => null
    };
    store = new Vuex.Store({
      getters
    });
    wrapper = shallowMount(Home, {
      stubs: mockCustomElements,
      store,
      localVue
    });
    expect(wrapper).toMatchSnapshot();
  });
});
