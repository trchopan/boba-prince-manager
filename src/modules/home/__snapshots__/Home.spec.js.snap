// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue matchs snapshot 1`] = `
<div>
  <my-header-stub>Home page</my-header-stub>
  <!---->
</div>
`;

exports[`Home.vue matchs snapshot 2`] = `
<div>
  <my-header-stub>Home page</my-header-stub>
  <div class="home-container">
    <p>Not yet have account?</p>
    <v-btn-stub to="/signup" outline="" color="secondary">Click here to sign up</v-btn-stub>
  </div>
</div>
`;
